### PR TITLE
Test with Java 17 in addition to Java 11 and Java 8

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Test Java 17 in additional to Java 11 and Java 8

Use Java 17 in the Jenkinsfile to compile and test.

Tests are failing with a serialization error.  I need more guidance on the changes needed to fix the failing tests.

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/jenkinsci/git-plugin/blob/master/CONTRIBUTING.adoc) doc
- [x] Unit tests pass locally with my changes
- [x] No Javadoc warnings were introduced with my changes
- [x] No spotbugs warnings were introduced with my changes
- [x] Documentation in README has been updated as necessary
- [x] Online help has been added and reviewed for any new or modified fields
- [x] I have interactively tested my changes
- [x] Any dependent changes have been merged and published in upstream modules (like git-client-plugin)

## Types of changes

- [x] Dependency or infrastructure update

### CODE REVIEW

**Comment by jglick** [comment]
Running locally I do see

```
java.io.IOException: java.lang.RuntimeException: Failed to serialize hudson.model.AbstractProject#scm for class hudson.model.FreeStyleProject
	at hudson.XmlFile.write(XmlFile.java:220)
	at hudson.model.AbstractItem.save(AbstractItem.java:617)
	at hudson.model.Job.save(Job.java:193)
	at hudson.model.AbstractProject.save(AbstractProject.java:289)
	at hudson.model.AbstractProject.setScm(AbstractProject.java:1506)
	at hudson.plugins.git.AbstractGitProject.setupProject(AbstractGitProject.java:193)
	at hudson.plugins.git.SCMTriggerTest.check(SCMTriggerTest.java:244)
	at hudson.plugins.git.SCMTriggerTest.testNamespaces_with_refsHeadsMaster(SCMTriggerTest.java:63)
	at …
Caused by: java.lang.RuntimeException: Failed to serialize hudson.model.AbstractProject#scm for class hudson.model.FreeStyleProject
	at …
Caused by: java.lang.RuntimeException: Failed to serialize hudson.plugins.git.GitSCM#userRemoteConfigs for class hudson.plugins.git.GitSCM
	at …
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field protected transient int java.util.AbstractList.modCount accessible: module java.base does not "opens java.util" to unnamed module @38a8f1a9
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
	at com.thoughtworks.xstream.converters.reflection.FieldDictionary.buildDictionaryEntryForClass(FieldDictionary.java:176)
	at com.thoughtworks.xstream.converters.reflection.FieldDictionary.buildMap(FieldDictionary.java:142)
	at com.thoughtworks.xstream.converters.reflection.FieldDictionary.fieldsFor(FieldDictionary.java:80)
	at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:167)
	at hudson.util.RobustReflectionConverter.doMarshal(RobustReflectionConverter.java:206)
	at …
```

I do see earlier in the build

```
[INFO] Setting jenkins.addOpens to --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.io=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED
```

as expected. The test passes with

```diff
diff --git pom.xml pom.xml
index 34e29fa1..afe0d153 100644
--- pom.xml
+++ pom.xml
@@ -62,7 +62,6 @@
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <argLine>-Dfile.encoding=${project.build.sourceEncoding}</argLine>
     <jenkins.version>2.289.1</jenkins.version>
     <no-test-jar>false</no-test-jar>
     <useBeta>true</useBeta>
```

Not sure why https://github.com/jenkinsci/git-plugin/blob/1435a5ec0941a1f79386e55384d6e6fed7ea59a6/pom.xml#L64-L65 are even here to begin with. Added in #862 for no obvious reason. I think both could be deleted.

@basil is it known that an explicit `argLine` is not compatible with the Java 17 changes as of https://github.com/jenkinsci/plugin-pom/pull/524? As in https://github.com/jenkinsci/plugin-pom/pull/453 I think the idea is that plugins should configure the Surefire mojo where needed to add system properties, rather than overriding this property. If confirmed, should be mentioned in https://github.com/jenkinsci/plugin-pom/releases/tag/plugin-4.40.

**Comment by jglick** [comment]
(Or we could introduce a property like `surefire.argLine.basic` with default JVM settings that you could override, and include that in `argLine`, or better yet directly in Surefire `<configuration>`.)

File: Jenkinsfile [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 buildPlugin(failFast: false,
             configurations: [
-                [platform: 'linux', jdk: '11'],
-                [platform: 'windows', jdk: '8'],
+                [platform: 'linux',   jdk: '11', jenkins: '2.342'],
```
[the remark this answers is not in the document]
Comment (MarkEWaite) [reply Jenkinsfile]: :)  Thanks for catching that.  Made the same change in multiple repositories and failed to finish the job in this one.

Fixed the mistake in c8e4e1f554362bbe65d2f1199a7341608871db77

**Comment by MarkEWaite** [comment]
Also removed the source encoding setting without seeing any negative affect on compile time messages.  See e711ad8974458f80573223b0e13a05b7bcf8d2eb

**Pull request merged**